src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -163,22 +163,23 @@ private async Task<ObservedDataMapping> copyObservedData(ObservedDataMapping obs
          };
       }
 
-      private Task<Section[]> copySectionContents(Section[] sections)
+      private Section[] copySectionContents(Section[] sections)
       {
+         // Not in parallel for now to ensure that we do not override files already downloaded as we are checking for file existence
          if (sections == null)
-            return Task.FromResult(Array.Empty<Section>());
+            return Array.Empty<Section>();
 
-         return Task.WhenAll(sections.Select(copySectionContent));
+         return sections.Select(copySectionContent).ToArray();

```

Comment:
Content files were being downloaded in //. With very small files used multiple times, this would create race conditions.